GravatarApp/Networking/GravatarURLSession.swift

```diff
@@ -0,0 +1,40 @@
+import Foundation
+import Gravatar
+
+/// Intercepts the network requests to perform common operations.
+final class GravatarURLSession: URLSessionProtocol, Sendable {
+    static let shared = GravatarURLSession()
```

Comment:
Just curious. Any particular reason to make a shared instance instead of instantiate a new one when needed?

Reply:
Do you mean we could have multiple URLSessions? Because we wouldn't want that. URLSession should manage the requests and optimize them while letting them share the same cookies, cache etc.

We only want one `URLSession` instance so it's by nature a singleton. 

Or do you mean we could make *only* the `urlSession` static and create a new GravatarURLSession() everywhere?

Then it just *hides* that there's a single URLSession instance inside. Can even be confusing sometimes.